### PR TITLE
Display Hearings Scheduled column in the landing page

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves https://vajira.max.gov/browse/CASEFLOW-1042

### Description
- batch query vacols and postress to determine the slot for each hearing day
- feature toggle the display and download of this column - [slack convo](https://dsva.slack.com/archives/C01MBDKAC7P/p1625582263061300?thread_ts=1625089585.054000&cid=C01MBDKAC7P)

### Acceptance Criteria (Reviewers should check these off)
- [x] Hearings schedule includes a new "Hearings Scheduled" column displaying number of hearings scheduled and total number of slots for each hearing day (e.g., "5 of 8")
- [x] When user clicks "Download current view", the column is also included in the downloaded CSV
- [x] 🧘‍♂️ Figure out why user timing event isn't sent to Google Analytics that measures load time of hearing days in the list
Looks like we're using ApiUtil, and I'd expect this code to send a timing event for the POST: https://github.com/department-of-veterans-affairs/caseflow/blob/ed0a3fcf3f565167329be37d7652046f5465ccbd/client/app/util/ApiUtil.js#L16

### Testing Plan
1. Login as `BVASYELLOW` and to go the http://localhost:3000/hearings/schedule. Ensure the new column does **not** display on page and does **not**  appear on the CSV
2. on Console, run `FeatureToggle.enable!(:view_and_download_hearing_scheduled_column)`
1.  Login as `BVASYELLOW` and to go the http://localhost:3000/hearings/schedule. Ensure the new column **does** on page and does appear on the CSV

### User Facing Changes
 BEFORE|AFTER
 ---|---
Screen Shot 2021-07-06 at 12 04 24 PM|Screen Shot 2021-07-06 at 12 03 38 PM

### Release Plans
1. Turn Feature toggle on for Dave
3. Deploy this branch to PROD

### CODE REVIEW

File: client/app/hearings/constants.js [hunk 1]
```diff
@@ -141,3 +141,8 @@ export const REQUEST_TYPE_OPTIONS = [
   { label: 'Virtual',
     value: HEARING_REQUEST_TYPES.virtual }
 ];
+
+export const ENDPOINT_NAMES = {
```
Comment (rubaiyat22) [inline client/app/hearings/constants.js:145]: Per AC
>Figure out why user timing event isn't sent to Google Analytics that measures load time of hearing days in the list

File: client/app/hearings/containers/AssignHearingsContainer.jsx [hunk 2]
```diff
@@ -61,7 +62,8 @@ class AssignHearingsContainer extends React.PureComponent {
 
     const requestUrl = `/hearings/schedule/assign/hearing_days?regional_office=${selectedRegionalOffice}`;
 
-    return ApiUtil.get(requestUrl, { timeout: { response: getMinutesToMilliseconds(5) } }
+    return ApiUtil.get(requestUrl, { timeout: { response: getMinutesToMilliseconds(5) } },
```
Comment (rubaiyat22) [inline client/app/hearings/containers/AssignHearingsContainer.jsx:65]: per AC
>Figure out why user timing event isn't sent to Google Analytics that measures load time of hearing days in the list

File: client/app/hearings/containers/ListScheduleContainer.jsx [hunk 3]
```diff
@@ -89,7 +89,7 @@ export class ListScheduleContainer extends React.Component {
       timeout: { response: getMinutesToMilliseconds(2) }
     };
 
-    return ApiUtil.get(requestUrl, requestOptions).then((response) => {
+    return ApiUtil.get(requestUrl, requestOptions, ENDPOINT_NAMES.HEARINGS_SCHEDULE).then((response) => {
```
Comment (rubaiyat22) [inline client/app/hearings/containers/ListScheduleContainer.jsx:92]: per AC
> Figure out why user timing event isn't sent to Google Analytics that measures load time of hearing days in the list

File: app/controllers/hearings/hearing_day_controller.rb [hunk 4]
```diff
@@ -17,7 +17,8 @@ def index
 
       format.json do
         if hearing_day_range.valid?
-          serialized_hearing_days = ::HearingDaySerializer.serialize_collection(hearing_days_in_range_for_user)
+          serialized_hearing_days =
+            ::HearingDaySerializer.serialize_collection(hearing_days_in_range_for_user, current_user)
```
Comment (rubaiyat22) [inline app/controllers/hearings/hearing_day_controller.rb:21]: Current user is needed here to determine if the feature toggle is turned on for them

File: app/queries/hearing_day_filled_slots_query.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+class HearingDayFilledSlotsQuery
+  def initialize(hearing_days)
+    @hearing_days = hearing_days
+  end
+
+  def call
+    result = ama_hearings_count_per_day
+    legacy_hearings_count_per_day.each do |hearing_day_id, filled_slots_count|
+      result[hearing_day_id.to_i] = 0 if result[hearing_day_id.to_i].blank?
```
Comment (rubaiyat22) [inline app/queries/hearing_day_filled_slots_query.rb:18]: `legacy_hearings_count_per_day` returns string keys so they are converted to integer here.

File: app/models/vacols/case_hearing.rb [hunk 6]
```diff
@@ -38,6 +38,8 @@ class VACOLS::CaseHearing < VACOLS::Record
     E: Constants.HEARING_DISPOSITION_TYPES.scheduled_in_error
   }.freeze
 
+  CLOSED_HEARING_DISPOSITIONS = [:C, :P, :E].freeze
```
Comment (sahalliburton) [inline app/models/vacols/case_hearing.rb:41]: Not super necessary, but it would be great to have a comment here explaining what these are referring to

Comment (rubaiyat22) [reply app/models/vacols/case_hearing.rb:42]: Clarified in this 3d53a14

**Review by tomas-nava — APPROVED** [review]
approve!

with non-blocking suggestions

File: app/queries/hearing_day_filled_slots_query.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+      result[hearing_day_id.to_i] += filled_slots_count
+    end
+
+    result
+  end
+
```
[the remark this answers is not in the document]
Comment (rubaiyat22) [reply app/queries/hearing_day_filled_slots_query.rb:23]: updated in 1ba5487

**Pull request merged**